src/wallet/wallet.cpp

```diff
@@ -168,7 +168,7 @@ bool CWallet::AddSaplingZKey(
 
     if (!IsCrypted()) {
         auto ivk = sk.expsk.full_viewing_key().in_viewing_key();
-        return CWalletDB(strWalletFile).WriteSaplingZKey(ivk,sk, mapSaplingZKeyMetadata[ivk],defaultAddr); 
+        return CWalletDB(strWalletFile).WriteSaplingZKey(defaultAddr.get(),ivk,sk, mapSaplingZKeyMetadata[ivk]); 
```

Comment:
defaultAddr is no longer optional and the .get() needs to be removed from this method.
Edit: and similar methods